pulsar-broker/src/main/java/org/apache/pulsar/broker/service/DrainingHashesTracker.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.service;
+
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * A thread-safe map to store draining hashes in the consumer.
+ */
+@Slf4j
+public class DrainingHashesTracker {
+    private final String dispatcherName;
+    private final UnblockingHandler unblockingHandler;
+    // optimize the memory consumption of the map by using primitive int keys
+    private final Int2ObjectOpenHashMap<DrainingHashEntry> drainingHashes = new Int2ObjectOpenHashMap<>();
+    int batchLevel;
+    boolean unblockedWhileBatching;
+
+    /**
+     * Represents an entry in the draining hashes tracker.
+     */
+    public static class DrainingHashEntry {
+        private final Consumer consumer;
+        private int refCount;
+        private int blockedCount;
+
+        /**
+         * Constructs a new DrainingHashEntry with the specified consumer ID.
+         *
+         * @param consumer the ID of the consumer
+         */
+        DrainingHashEntry(Consumer consumer) {
+            this.consumer = consumer;
+        }
+
+        /**
+         * Gets the consumer ID.
+         *
+         * @return the consumer ID
+         */
+        public Consumer getConsumer() {
+            return consumer;
+        }
```

Comment:
Does the Javadoc need to be updated to reflect the changes in https://github.com/apache/pulsar/commit/e9fa63f2f77d300e899c6b4faa4d411893dcdbdd (use Consumer instance instead of consumer id)?

Reply:
thanks @phil-cd , addressed in 46209d8e016db22ea72ab67c931b2e85be4274cc